File: typescript/cmd/aps-adapter.ts
#!/usr/bin/env node
// Usage: aps-adapter <test-cmd> [test-args...]
// All argv past argv[2] is the project's test command. The adapter sets
// APS_IR_PATH for each mutator job. This positional form survives
// `gherkin-mutator --runner-worker`, which splits on whitespace.
import { serve } from "../src/adapter.js";

function usage(): never {
  process.stderr.write("usage: aps-adapter <test-cmd> [test-args...]\n");
  process.exit(2);
}

async function main(argv: string[]): Promise<number> {
  const args = argv.slice(2);
  if (args.length === 0 || args[0] === "-h" || args[0] === "--help") {
    if (args.length === 0) usage();
    process.stderr.write("usage: aps-adapter <test-cmd> [test-args...]\n");
    return 0;
  }
  const [command, ...cmdArgs] = args;
  await serve({ command, args: cmdArgs });
  return 0;
}

main(process.argv).then((code) => process.exit(code));

File: rust/aps-kit/src/bin/aps-adapter.rs
// aps-adapter: APS-conformant runner adapter for Rust projects.
//
// Usage:
//     aps-adapter <test-cmd> [test-args...]
use std::process::ExitCode;

use aps_kit::cli::run_adapter;

fn main() -> ExitCode {
    let args: Vec<String> = std::env::args().skip(1).collect();
    let stdin = std::io::stdin();
    let stdout = std::io::stdout();
    let stderr = std::io::stderr();
    let code = run_adapter(&args, stdin.lock(), stdout.lock(), stderr.lock());
    ExitCode::from(code as u8)
}
